### PR TITLE
[Tests-Only] Delete groups in after scenario only if it exists and possible to delete

### LINKED ISSUE #37417: Acceptance tests Illegal string offset
https://drone.owncloud.com/owncloud/user_ldap/2518/28/11
```
Warning: Illegal string offset 'shouldExist' in /var/www/owncloud/testrunner/tests/acceptance/features/bootstrap/Provisioning.php line 4002
```

Seems to happen when running some app tests - e.g quite a few user_ldap pipelines in that drone CI run.

It probably happened with the recent changes in PR #37406

### PR DESCRIPTION
**Pull request by kiranparajuli589**
## Description
Remove groups in after scenario only if it exists and possible to delete

## Related Issue

- Fixes https://github.com/owncloud/core/issues/37417

## How Has This Been Tested?

- :robot: 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: tests/acceptance/features/bootstrap/Provisioning.php [hunk 1]
```diff
@@ -3999,13 +3999,19 @@ public function cleanupDatabaseGroups() {
 		$previousServer = $this->currentServer;
 		$this->usingServer('LOCAL');
 		foreach ($this->createdGroups as $group => $groupData) {
-			if ($groupData["shouldExist"] and $groupData["possibleToDelete"]) {
+			if (
+				isset($groupData["shouldExist"]) and $group["shouldExist"]
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/Provisioning.php:4003]:
```suggestion
				isset($groupData["shouldExist"]) and $groupData["shouldExist"]
```

File: tests/acceptance/features/bootstrap/Provisioning.php [hunk 2]
```diff
@@ -3999,13 +3999,19 @@ public function cleanupDatabaseGroups() {
...
+				and $groupData["possibleToDelete"]
+			) {
 				$this->cleanupGroup($group);
 			}
 		}
 		$this->usingServer('REMOTE');
 		foreach ($this->createdRemoteGroups as $remoteGroup => $groupData) {
-			if ($groupData["shouldExist"] and $groupData["possibleToDelete"]) {
+			if (
+				isset($groupData["shouldExist"]) and $group["shouldExist"]
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/Provisioning.php:4012]:
```suggestion
				isset($groupData["shouldExist"]) and $groupData["shouldExist"]
```

**Review by phil-davis — COMMENTED** [review]
Also we do not normally use `and` - we use `&&` - so you could change that.
And make sure to `make test-php-style` to check the code style before pushing.

**Comment by kiranparajuli589** [comment]
@phil-davis addressed your comments. CI does not start currently :-(

**Review by phil-davis — APPROVED** [review]

**Comment by phil-davis** [comment]
drone has caught up with the backlog. This can be merged when CI result comes.

**Pull request merged**